src/parser/jobs/mnk/modules/Fists.tsx

```diff
@@ -190,52 +196,68 @@ export class Fists extends Analyser {
 				value,
 				color: this.chartColours[id],
 				columns: [
-					this.getFistName(id, unknownFist),
+					this.getFistName(id),
 					this.parser.formatDuration(value),
-					this.getFistUptimePercent(id, unknownFist) + '%',
+					this.getFistUptimePercent(id) + '%',
 				] as const,
 			}
 		}).filter(datum => datum.value > 0)
 
+		this.debug(uptimeKeys)
+
 		this.statistics.add(new PieChartStatistic({
 			headings: ['Fist', 'Uptime', '%'],
 			data,
 		}))
 	}
 
-	getFistGCDCount(fistId: number): number {
+	getFistGCDCount(fistId: FistId): number {
 		return this.fistory
 			.filter(fist => fist.id === fistId)
 			.reduce((total, current) => total + current.gcdCounter, 0)
 	}
 
-	getFistUptimePercent(fistId: number, unknownFist: boolean): string {
-		const status = this.data.getStatus(fistId)
-		if (status == null) { return '0' }
+	getFistUptimePercent(fistId: FistId): string {
+		const fightUptime = this.parser.currentDuration - this.invulnerability.getDuration({types: ['invulnerable']})
```

Comment:
Typically Downtime is the preferred method to do uptime refunds, since it also accounts for time spent dead (so that uptime calcs don't double-ding against the "don't die" recommendation).

Reply:
File removed in 6.0 anyway since Fists are gone, leaving this alone for 5.x since it's about status uptime, not if that uptime was useful. Realistically this chart returns 100% pretty much exclusively, since there's no reason to use other fists anymore.